cpplint.py

```diff
@@ -934,9 +934,93 @@ def unicode_escape_decode(x):
 # This is set by --headers flag.
 _hpp_headers = set([])
 
-# {str, bool}: a map from error categories to booleans which indicate if the
-# category should be suppressed for every line.
-_global_error_suppressions = {}
+class ErrorSuppressions:
+  """Class to track all error suppressions for cpplint"""
+
+  class SuppressionRange:
+    """Class to represent a range of line numbers for which an error is suppressed"""
+    def __init__(self, begin=0, end=math.inf):
+      self._begin = begin
+      self._end = end
+
+    def __str__(self):
+      return f'[{self.begin}-{self.end}]'
+
+    def __contains__(self, obj):
+      if isinstance(obj, ErrorSuppressions.SuppressionRange):
+        return self.begin <= obj.begin and self.end >= obj.end
+      elif isinstance(obj, int):
+        return self.begin <= obj <= self.end
+      raise TypeError()
+
+    @property
+    def begin(self):
+      return self._begin
+
+    @property
+    def end(self):
+      return self._end
+
+    @end.setter
+    def end(self, end):
+      assert end > self.begin
+      self._end = end
+
+  def __init__(self):
+    self._suppressions = {}
```

Comment:
nit: possibly a defaultdict would make the code later easier to read/write, not having to worry about None results

Reply:
I updated to use defaultdict